### PR TITLE
Load content runtime data 'on demand'

### PR DESCRIPTION
**Pull request by jdgleaver**
## Description

At present, when playtime logging is enabled, the runtime information associated with each entry of a playlist is loaded when the playlist itself is opened. On the vast majority of platforms (and all desktop systems) the performance impact of this is negligible - but on devices with very slow storage (typically Nintendo consoles), once a playlist accumulates ~100 runtime logs then the added delay when opening the playlist becomes noticeable (and annoying).

With this PR, runtime logs are only read when the menu driver presents the associated entries (the values are cached, and only reloaded once the playlist is opened anew). This places an upper limit on the number of runtime logs that can be loaded at any one time:

- For RGUI, this is 17

- For XMB, this is ~11

- For Ozone, this is 1 (Ozone doesn't use playlist sublabels, so it can handle this more efficiently)

- GLUI unfortunately loops through all entries regardless of what is shown, so no benefit is seen here...

Basically, this means that playlists can be arbitrarily long, and have an arbitrary number of runtime logs, and performance will not suffer on weak devices (unless they use GLUI, but this is not available on console platforms anyway).

Note that further improvements are possible with changes to the menu entry code - I'll look into this later.

### CODE REVIEW

**Comment by bparker06** [comment]
This pull request **introduces 1 alert** when merging a4785b757ff98202fb70ab23c7d6a6df5dc0ae8c into ba3f164bdd06b4e00b4712e0d5c1aed5a5aac51e - [view on LGTM.com](https://lgtm.com/projects/g/libretro/RetroArch/rev/pr-481cab55135aba6962a5e54922ea3273233c3267)

**new alerts:**

* 1 for Declaration hides parameter

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jdgleaver** [comment]
Okay, just pushed a commit to get rid of that LGTM warning...
